### PR TITLE
[MWPW-152181] Tabbing gets "stuck" after second instance of merch-card CTAs, within a segment blade

### PR DESCRIPTION
**Pull request by rohitsahu**
Allow the default tab action to proceed if the current focused element is from the last merch card.

Resolves: [MWPW-152181](https://jira.corp.adobe.com/browse/MWPW-152181)

Test URLs:
Before: https://main--milo--adobecom.hlx.page/drafts/rosahu/rosahu
After: https://mwpw-152181-2--milo--rohitsahu.hlx.page/drafts/rosahu/rosahu

### CODE REVIEW

**Review by mokimo — APPROVED** [review]

**Review by 3ch023 — CHANGES_REQUESTED** [review]
Hi @rosahu /libs/features/mas/* are source files, you need to include also generated dependencies. pages are loading files from /deps/mas/*, so those should be updated and included in the PR

```
cd libs/features/mas 
npm run build
```

should do the trick.
I noticed that from /web-components 'npm run build' seems to not work properly, maybe you could fix that? but it's an optional request.

**Comment by rohitsahu** [comment]
> Hi @RosaHu /libs/features/mas/* are source files, you need to include also generated dependencies. pages are loading files from /deps/mas/*, so those should be updated and included in the PR
> 
> ```
> cd libs/features/mas 
> npm run build
> ```
> 
> should do the trick. I noticed that from /web-components 'npm run build' seems to not work properly, maybe you could fix that? but it's an optional request.

Done.
Also, "npm run build" work fine from web-components as well

File: libs/deps/mas/merch-card-all.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+\f\r]`,M=/<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z][^>\s]*)|(\/?$))/g,Le=/-->/g,Ue=/>/g,_=RegExp(`>|${se}(?:([^\\s"'>=/]+)(${se}*=${se}*(?:[^
```
[the remark this answers is not in the document]
Comment (rohitsahu) [reply libs/deps/mas/merch-card-all.js:2]: @3ch023  not sure if I should try to manually fix this vulnerability. What's the workaround?

**Comment by 3ch023** [comment]
Hi @rohitsahu I tried to test it on the prod page where the bug was reported and it seems to still not work with your changes, could you please take a look?
https://main--cc--adobecom.hlx.live/products/photoshop/free-trial-download?milolibs=mwpw-152181-2--milo--rohitsahu

**Review by 3ch023 — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Comment by Priyankavgit** [comment]
I've validated the issue on the Milo pages using the fork branch, and the issue has been resolved. The pull requests have been updated with the 'Ready for Stage' label.
Jira Ticket: https://jira.corp.adobe.com/browse/MWPW-152181

**Review by overmyheadandbody — COMMENTED** [review]
I'm not sure I'm getting this. On https://mwpw-152181-2--milo--rohitsahu.hlx.page/drafts/rosahu/rosahu, pressing Tab after the last element of a merch card doesn't do anything. For example, in this screenshot, the focused element is the "Buy Now" one from the first card. Pressing Tab has no effect, the focus remains on that element. Pressing Shift + Tab focuses the previous link, which is correct. My expectation would be that pressing Tab in this scenario would focus the link from "This is the green promo text with link" from the next card.
Screenshot 2024-08-02 at 16 29 45

I'm not requesting changes, since that's the current behavior on production as well, but to me it looks like there's still an issue to be solved. Am I testing in the right place? Is this another bug that needs to be treated separately?

**Pull request merged**